CMakeLists.txt

```diff
@@ -2265,12 +2265,17 @@ if(COREAUDIO)
   target_include_directories(mixxx-lib SYSTEM PUBLIC lib/apple)
 endif()
 
+
 # FAAD AAC audio file decoder plugin
 find_package(MP4)
 find_package(MP4v2)
 # It is enabled by default on Linux only, because other targets have other
 # solutions. It requires MP4 or MP4v2.
-default_option(FAAD "FAAD AAC audio file decoder support" "UNIX;NOT APPLE;(MP4_FOUND OR MP4v2_FOUND)")
+if (UNIX AND NOT APPLE AND (MP4_FOUND OR MP4v2_FOUND))
```

Comment:
Why did you change this and replace it with code that contains duplicated strings?

Reply:
"not working" only in the commit message is not a valid reason for introducing an ugly workaround.

Reply:
I tested this be removing the development header of mp4v2 and the default_option works as expected by setting FAAD:BOOL=OFF.

Reply:
It does not work on the PPA, our Bionic build has no m4a playback even if libfaad is enabled. It took me hours to test various alternatives, and I came up with this as the only one working.
I guess it is a cmake version issue. I will add a comment. 
If you have interest to improve this further, please propose an alternative.

Reply:
As expected. We now we need to add hacks to our CMake build, because the Debian/Ubuntu packaging is not cleanly separated and doesn't cope with different configurations properly.

Reply:
Our CI builds prove that this is NOT a build issue.

Reply:
Our CI builds prove that this is NOT a build issue. They work fine for both Ubuntu 18.04 and 20.04 with the desired configurations.

Reply:
I have checked the CI build. We are building 18.04 with -DFAAD=ON without any problems. So please clarify what's the actual issue and why it must be resolved here.

Reply:
Maybe we suffer a 3.16.3 bug solved in 3.16.9.  ..

Reply:
Ok, I noticed that we are using 3.16.x in the CI build without a specific minor version number.

Reply:
This change must immediately be reverted when merging into main!

Reply:
How about we restrict the CMake setup step to macOS? Ubuntu can use the cmake version from the repos and windows comes with its own cmake version as part of VS2019 anyway. Then we notice this stuff in CI.

Reply:
Yes, that's IMHO the way to go. 

This works flawlessly for main, because we have dropped Ubuntu Bionic there.

For 2.3 we need to update cmake. But we can do it via a additional PPA added to debian_buildenv.sh just like on our PPA. Is it worth the work before releasing?

Reply:
Let's do this in follow up PR.

Reply:
> Maybe we suffer a 3.16.3 bug solved in 3.16.9. ..

Side note: I don't get why the version is so old. Why didn't they update? I thought stable distros only lock major updates, but delay bugfix releases.